maven-embedder/src/main/java/org/apache/maven/cli/internal/BootstrapCoreExtensionManager.java

```
@@ -114,14 +119,38 @@ private CoreExtensionEntry createExtension( CoreExtension extension, List<Artifa
     {
         String realmId =
             "coreExtension>" + extension.getGroupId() + ":" + extension.getArtifactId() + ":" + extension.getVersion();
-        ClassRealm realm = classWorld.newRealm( realmId, null );
-        log.debug( "Populating class realm " + realm.getId() );
-        realm.setParentRealm( parentRealm );
+        ClassRealm realm;
+        Set<String> providedArtifacts = Collections.emptySet();
+        if ( "parent-first".equals( extension.getClassLoadingStrategy() ) )
```

Should it fail if the value is unknown? `classLoadingStrategy`